test/e2e/plugin/verify_signature.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/notaryproject/notation-go/plugin/proto"
+	"github.com/notaryproject/notation-go/verifier"
+	"github.com/notaryproject/notation/test/e2e/plugin/internal/io"
+	"github.com/spf13/cobra"
+)
+
+func verifySignatureCommand() *cobra.Command {
+	return &cobra.Command{
+		Use: "verify-signature",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			req := &proto.VerifySignatureRequest{}
+			if err := io.UnmarshalRequest(req); err != nil {
+				return &proto.RequestError{Code: proto.ErrorCodeValidation, Err: err}
+			}
+
+			if err := validateVerifySignatureRequest(*req); err != nil {
+				return &proto.RequestError{Code: proto.ErrorCodeValidation, Err: err}
+			}
+			return runVerifySignature(req)
+		},
+	}
+}
+
+func runVerifySignature(req *proto.VerifySignatureRequest) error {
+	return io.PrintResponse(extractVerificationResult(req))
+}
+
+// validateVerifySignatureRequest validates required field existence.
+func validateVerifySignatureRequest(req proto.VerifySignatureRequest) error {
+	// check req.Signature.CriticalAttributes
+	if err := validateRequiredField(req.Signature.CriticalAttributes, fieldSet(
+		"ContentType",
+		"SigningScheme",
+		"Expiry",
+		"AuthenticSigningTime",
+		"ExtendedAttributes")); err != nil {
+		return err
+	}
+
+	// check required extended attributes
+	if _, ok := req.Signature.CriticalAttributes.ExtendedAttributes[verifier.HeaderVerificationPlugin]; !ok {
+		return fmt.Errorf("missing extended attribute: %s", verifier.HeaderVerificationPlugin)
+	}
+	if _, ok := req.Signature.CriticalAttributes.ExtendedAttributes[verifier.HeaderVerificationPluginMinVersion]; !ok {
```

Comment:
HeaderVerificationPluginMinVersion is an optional field. Let's not fail in its absence.

Can we have a test that generates a signature with plugin min version as 2.0 and and verifies that signature against a verification plugin with version 1.0?

Reply:
Removed verifier.HeaderVerificationPlugin and verifier.HeaderVerificationPluginMinVersion check from the plugin, because those extended attributes will not be pass to plugin.

Also the plugin minimum version test cases are blocked by [feat: plugin version comparison functionality #237](https://github.com/notaryproject/notation-go/pull/237)

Reply:
Can you add an issue in notation repo to track completing these tests and link it here?

Reply:
Has this been created? I can't find the issue in https://github.com/notaryproject/notation/issues
Can you post the issue list here?

Also, let's resolve comments only after providing an update, otherwise it is hard to understand what has been done.

Reply:
I added an issue in notation-go https://github.com/notaryproject/notation-go/issues/244

https://github.com/notaryproject/notation-go/pull/237 has been merged. We can add more test cases for verification plugin version now.

Reply:
Added plugin minimum version test cases now.